crates/ruff_linter/src/rules/pep8_naming/rules/camelcase_imported_as_acronym.rs

```diff
@@ -97,3 +90,34 @@ pub(crate) fn camelcase_imported_as_acronym(
     }
     None
 }
+
+fn is_ignored_because_of_import_convention(
+    asname: &str,
+    stmt: &Stmt,
+    alias: &Alias,
+    checker: &Checker,
+) -> bool {
+    let full_name = if let Some(import_from) = stmt.as_import_from_stmt() {
+        // Never test relative imports for exclusion because we can't resolve the full-module name.
+        let Some(module) = import_from.module.as_ref() else {
+            return false;
+        };
+
+        if import_from.level != 0 {
+            return false;
+        }
```

Comment:
I'd prefer to assert the invariant here: `module` should never be non-`None` if `level == 0` (that would be invalid syntax)

```suggestion
        if import_from.level != 0 {
            return false;
        }
        let module = import_from
            .module
            .as_ref()
            .expect("Expected `module` to always be non-`None` for a non-relative `import from` statement");
```

Reply:
I prefer the more error resilient code in case we ever change how the parser recovers.